execution/market.go

```diff
@@ -111,9 +111,9 @@ var (
 type PriceMonitor interface {
 	OnTimeUpdate(now time.Time)
 	CheckPrice(ctx context.Context, as price.AuctionState, trades []*types.Trade, persistent bool) bool
-	GetCurrentBounds() []*types.PriceMonitoringBounds
+	GetCurrentBounds(inAuction bool) []*types.PriceMonitoringBounds
```

Comment:
Not really a fan of this, TBH. Isn't it more sensible to handle some of the work `OnTimeUpdate`? We might end up calling `GetCurrentBounds` or `GetValidPriceRange` several times in a single block, we don't have to check the validity of the bounds each time, just on time update.

Reply:
If we really do need to keep track of the auction state in the price monitor, I'd argue it makes more sense to inject the auction state, rather than pass in `m.as.InAuction()` as arguments, and passing the auction state as an argument to the liquidity engine constructor. I can see why price monitoring would need to access the auction state more readily than the liquidity engine...

Reply:
Yeah I didn't love it either, but wanted to make sure it always returns actual bounds (and not bounds that would turn out to not be "bitting" when push comes to shove as they've actually expired). 

Could add a method to price monitor that get's called when market gets put into and out of auction (this is somewhat missing anyway, for example we reactive the bounds when price monitoring engine is ready to end the auction, but there are actually no guarantees that market will indeed end an auction, so it'd help with that too). What do you think?

Reply:
Ok, your suggestion to just give price monitoring engine access to auctionState is probably the cleanest, will proceed on that basis.

Reply:
Addressed now I believe.